cmd/clusterawsadm/iamservice/service.go

```diff
@@ -0,0 +1,500 @@
+			}
+		}
+	}
+	err = attachRoleToInstanceProf(resource, client)
+	if err != nil {
+		return err
+	}
+	klog.V(2).Infof("created \"%s\" CAPA managed instance profile", res.InstanceProfileName)
+	return nil
+}
+
+// CreatePolicy creates a new CAPA Managed IAM Policy and attaches it to the roles as defined in the bootstrap configuration file.
+func CreatePolicy(resource go_cfn.Resource, tags map[string]string, client *iam.IAM) error {
+	res := resource.(*cfn_iam.ManagedPolicy)
+	tgs := []*iam.Tag{}
+	for k, v := range tags {
+		tag := iam.Tag{
+			Key:   aws.String(k),
+			Value: aws.String(v),
+		}
+		tgs = append(tgs, &tag)
+	}
+	rawdata := res.PolicyDocument.(*iamv1.PolicyDocument)
+	data, err := json.Marshal(rawdata)
+	if err != nil {
+		return errors.Wrapf(err, "corrupt policy document format for policy \"%s\"", res.ManagedPolicyName)
+	}
+	create, err := client.CreatePolicy(&iam.CreatePolicyInput{
+		Description:    &res.Description,
+		PolicyDocument: aws.String(string(data)),
+		PolicyName:     &res.ManagedPolicyName,
+		Tags:           tgs,
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case iam.ErrCodeEntityAlreadyExistsException:
+				klog.Warningf("policy \"%s\" already exists", res.ManagedPolicyName)
+				policies, err := listpolicies(client)
+				if err != nil {
+					return err
+				}
+				for _, policy := range policies {
+					if *policy.PolicyName == res.ManagedPolicyName {
+						return attachRolesToPolicy(policy, res.Roles, client)
+					}
+				}
+			default:
+				return errors.Wrapf(err, "failed to create CAPA managed IAM policy \"%s\"", res.ManagedPolicyName)
+			}
+		}
+	}
+	err = attachRolesToPolicy(create.Policy, res.Roles, client)
+	if err != nil {
+		return err
+	}
+	klog.V(2).Infof("created \"%s\" CAPA managed IAM policy", res.ManagedPolicyName)
+	return nil
+}
+
+func attachRoleToInstanceProf(resource go_cfn.Resource, client *iam.IAM) error {
+	res := resource.(*cfn_iam.InstanceProfile)
+	roleName, err := getRoleName(res.Roles[0])
+	if err != nil {
+		return err
+	}
+	_, err = client.AddRoleToInstanceProfile(&iam.AddRoleToInstanceProfileInput{
+		InstanceProfileName: &res.InstanceProfileName,
+		RoleName:            &roleName,
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case iam.ErrCodeEntityAlreadyExistsException, iam.ErrCodeLimitExceededException:
+				klog.Warningf("instance profile \"%s\" is already attached to its IAM role", res.InstanceProfileName)
+			default:
+				return errors.Wrapf(err, "failed to attach instance profile \"%s\" to IAM role \"%s\"", res.InstanceProfileName, roleName)
+			}
+		}
+	}
+	return nil
+}
+
+func attachPoliciesToRole(rolename *string, awsManagedPolicies []string, client *iam.IAM) error {
+	if awsManagedPolicies == nil {
+		// klog.Warningf("no policies defined to attach to the IAM role \"%s\"", *rolename) // TODO
```

Comment:
What's the next action here? Just enabling the warning, or something else?

Reply:
Reading further, looks like there's a question about outputting the ARNs elsewhere - is this the same?

Reply:
I think at that time I was wondering if we should issue a warning that the user hasn't defined any awsManagedPolicies in their CloudFormation config or just return back to parent call directly without any warnings.

Reply:
So should we give the warning?